testing/mocks/generic.go

```diff
@@ -81,20 +96,114 @@ var (
 		},
 	}
 
+	GenericEventTypes        = []flow.EventType{"deposit", "withdrawal"}
+	GenericCadenceEventTypes = []*cadence.EventType{
+		{
+			Location:            utils.TestLocation,
+			QualifiedIdentifier: "deposit",
+			Fields: []cadence.Field{
+				{
+					Identifier: "amount",
+					Type:       cadence.UInt64Type{},
+				},
+				{
+					Identifier: "address",
+					Type:       cadence.AddressType{},
+				},
+			},
+		},
+		{
+			Location:            utils.TestLocation,
+			QualifiedIdentifier: "withdrawal",
+			Fields: []cadence.Field{
+				{
+					Identifier: "amount",
+					Type:       cadence.UInt64Type{},
+				},
+				{
+					Identifier: "address",
+					Type:       cadence.AddressType{},
+				},
+			},
+		},
+	}
+	GenericCadenceEvents = []cadence.Event{
```

Comment:
As discussed, I would also make these functions. However, when we have types that depend on each other, I would rely on other generation functions:

```go
func GenericCadenceEventType(index int) cadence.EventType { ... }

func GenericCadenceEvent(index int) cadence.Event {
  ...WithType(GenericCadenceEventType(index))
}
```

This would make sure that event at index `1` would always correspond to type at index `1`. I'm not sure that ever breaks a test somewhere? WDYT?